test/mechanics/integrands/HistoryData.cpp

```diff
@@ -0,0 +1,397 @@
+#include "BoostUnitTest.h"
+
+#include "base/Group.h"
+#include "mechanics/cell/Cell.h"
+#include "mechanics/cell/CellInterface.h"
+#include "mechanics/cell/SimpleAssember.h"
+#include "mechanics/constitutive/laws/LinearElastic.h"
+#include "mechanics/constitutive/laws/MechanicsInterface.h"
+#include "mechanics/constraintsPde/Constraints.h"
+#include "mechanics/constraintsPde/ConstraintCompanion.h"
+#include "mechanics/dofs/DofNumbering.h"
+#include "mechanics/integrands/MomentumBalance.h"
+#include "mechanics/integrationtypes/IntegrationTypeTensorProduct.h"
+#include "mechanics/interpolation/InterpolationTrussLinear.h"
+#include "mechanics/mesh/MeshFem.h"
+#include "mechanics/mesh/MeshFemDofConvert.h"
+#include "mechanics/nodes/NodeSimple.h"
+
+#include <cassert>
+#include <functional>
+#include <map>
+#include <vector>
+
+using namespace Eigen;
+using namespace NuTo;
+using namespace NuTo::Groups;
+
+
+constexpr double SpecimenLength = 10.;
+constexpr unsigned int numElements = 4;
+
+//! @brief Standard history data management object.
+//! @tparam T: Data type of the history data
+template <typename T>
+class HistoryData
+{
+    std::map<int, std::vector<T>> mHisData;
+
+public:
+    //! @brief Returns the history data for a specific integration point
+    //! @param cellNum: Number of the cell containing the integration point
+    //! @param ipNum: Number of the integration point
+    //! @return History data of the integration point
+    const T& GetIPHistoryData(int cellNum, int ipNum) const
```

Comment:
what about `T& operator()(...)`?

Reply:
Doesn't work so well with inheritance. What would you expect, what the  function call `
mylaw(celldata, cellipdata);`will return? If we don't use inheritage it is an option.